client-library/src/honoMiddleware.ts

```diff
@@ -94,7 +96,15 @@ export function createHonoMiddleware<App extends HonoApp>(
 
     const service = env<FpxEnv>(c).FPX_SERVICE_NAME || "unknown";
 
-    const ctx = c.executionCtx;
+    const ctx =
+      getRuntimeKey() === "workerd"
```

Comment:
need to lookup if this is true for, e.g., vercel runtime

also - do we want to have separate middleware for different runtimes? might make sense from an observability perspective